src/common/connection.ts

```diff
@@ -432,44 +453,53 @@ class Connection extends EventEmitter {
     return this.request({command: 'ping'}).catch(() => this.reconnect())
   }
 
-  _whenReady<T>(promise: Promise<T>): Promise<T> {
+  /**
+   * Wait for a valid connection before resolving. Useful for deferring methods
+   * until a connection has been established.
+   */
+  private _waitForReady(): Promise<void> {
     return new Promise((resolve, reject) => {
-      promise.catch(reject)
       if (!this._shouldBeConnected) {
         reject(new NotConnectedError())
       } else if (this._state === WebSocket.OPEN && this._isReady) {
-        promise.then(resolve, reject)
+        resolve()
       } else {
-        this.once('connected', () => promise.then(resolve, reject))
+        this.once('connected', () => resolve())
       }
     })
   }
 
-  getLedgerVersion(): Promise<number> {
-    return this._whenReady(Promise.resolve(this._ledgerVersion!))
+  async getLedgerVersion(): Promise<number> {
```

Comment:
possible nit: missing an access modifier here. Is this on purpose?

Reply:
Which access modifier were you expecting? Since it's already a public method, we don't want to mark it `private` now and force a breaking change unnecessarily.

Reply:
I feel like my `eslint` config always yells at me to make access modifies explicit.  I prefer to make things explicit, though if this library doesn't then I defer to the existing style.

Reply:
Oh like adding "public"? I've actually never used that before. Not against it, but yea lets hold off to a future PR if you're interested in making that change. This library has only used them in 1-2 places and I see our eslint passing clean.